### PR TITLE
Helping with Colorado regional

### PR DESCRIPTION
**Pull request by Jnesselr**

### CODE REVIEW

**Review by awfulcarpet — APPROVED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -108,15 +110,21 @@ private void configureBindings() {
...
-        driverController.rightBumper().onTrue(speakerShoot);
-        driverController.x().onTrue(trapShoot);
-        driverController.y().onTrue(stageSpeakerShoot);
-        driverController.a().onTrue(intakeSubsystem.eject(2)).onFalse(intakeSubsystem.eject(0));
-        driverController.b().onTrue(longShot);
-        driverController.rightTrigger().onTrue(shootCommand);
-        driverController.leftTrigger().onTrue(stopShoot);
+        driverController.leftBumper().and(shooterSubsystem::isNoteInShooter).onTrue(ampShoot);
+        driverController.rightBumper().and(shooterSubsystem::isNoteInShooter).onTrue(speakerShoot);
+        driverController.leftTrigger().and(shooterSubsystem::isNoteInShooter).onTrue(stopShoot);
```
Comment (centiwati) [inline src/main/java/frc/robot/RobotContainer.java]: Do we need to have the note in the shooter to stop the shoot? That seems like possibly a bad idea.

File: src/main/java/frc/robot/commands/CommandHolder.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        if (holdCommand == null) {
+            holdCommand = new HoldCommand(intake);
+        }
+        return holdCommand;
+    }
+
+    public RumbleCommand rumbleCommand(RumbleCommand.Power power, RumbleCommand.Time time) {
+        return new RumbleCommand(controller, power, time);
+    }
+}
```
Comment (centiwati) [inline src/main/java/frc/robot/commands/CommandHolder.java:85]: I'm not entirely sure what this whole command is doing... can you explain it to me when you get a chance?

**Review by centiwati — APPROVED** [review]
Looks good enough. There was a change I believe that I thought you could make but it's not a dealbreaker and you probably did it for a reason.

**Review by Stubbynarwhal — APPROVED** [review]
I actually understood it (mostly) !! 🤩

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (GucciBanana11) [reply src/main/java/frc/robot/RobotContainer.java]: Yeah I got caught here @Jnesselr as well

Comment (Jnesselr) [reply src/main/java/frc/robot/RobotContainer.java]: Nope that's a good call. Probably not.

File: src/main/java/frc/robot/commands/CommandHolder.java (hunk 2 above)
Comment (Jnesselr) [reply src/main/java/frc/robot/commands/CommandHolder.java:85]: It basically handles the rumble on the controller and it only rumbles for a certain amount of time. The command handles the intensity and timer aspects of the RumbleCommand.

**Pull request merged**